actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserConsentActor.java

```diff
@@ -63,19 +86,81 @@ private void getUserConsent(Request request) {
     }
 
     private void updateUserConsent(Request request) {
+        Map<String, Object> consent = (Map<String, Object>) request.getRequest().getOrDefault(JsonKey.CONSENT, new HashMap<String, Object>());
+        String userId = (String) consent.getOrDefault(JsonKey.USER_ID, "");
+        if(StringUtils.isEmpty(userId)){
+            userId = (String) request.getContext().get(JsonKey.REQUESTED_BY);
+        }
+        String managedFor = (String)request.getContext().get(JsonKey.MANAGED_FOR);
+        if(StringUtils.isNotEmpty(managedFor)){
+            userId = managedFor;
```

Comment:
looks there is a possibility for manager_for to always override user. Imagine if this header was present (un-necessarily). Better to convert to if-else, I think.

Reply:
userId is not mandatory in request. So if managedFor is present that should get preference rite?